sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -120,22 +120,54 @@ abstract class LogicalPlan
   /**
    * Resolve the given `name` string against the given attribute, returning either 0 or 1 match.
```

Comment:
can you briefly explain the new resolution logic? I feel it's a little convoluted now as we are more likely to be ambiguous.

Reply:
The resolution logic needs to account to compare the full qualifier.   If you are trying to resolve db1.t1.i1, the match needs to happen against the entire qualifier of the attribute.   

For e.g: 
```SQL
 select db1.t1.i1 from t1   // current database is db1  and lets say t1 has  columns i1, i2. 
```
When resolving db1.t1.i1,  against the following set of Input attributes.   
we will have the following attribute references:
- AttributeReference ( qualifier Seq(db1, t1)  and name is i1)
- AttributeReference ( qualifier Seq(db1, t1) and name is i2) 

So the new resolution logic will match the entire qualifier sequence and then match the column ( ie the attribute name)

Reply:
At a high level the changes include: 
* The column resolution needs to account for the qualifier that is now a sequence of string. 
* Enhance the lookupRelation to add the qualifier appropriately. Right now, the qualifier used by the LogicalPlan nodes representing the relation leafnodes use only the table name as the qualifier when no alias is provided. This needs to be updated to add the database name and the table name for permanent views and tables and global temporary views. 

 Full details of the design and the resolution logic is in the [doc here](https://drive.google.com/file/d/1zKm3aNZ3DpsqIuoMvRsf0kkDkXsAasxH/view) in Section 3.3 and 3.4. 

So, Section 2 in the design doc has a lot of scenarios that were tested against different db’s and Spark and the expected behavior with the design here.   

If there is a specific scenario you have in mind that would be valid but get flagged as ambiguous, can you please let me know.   I’d be glad to look into it.